src/object.d

```diff
@@ -3241,181 +3241,105 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+// Compare class and interface objects for ordering.
+private int __cmp(Obj)(Obj lhs, Obj rhs)
+if (is(Obj : Object))
+{
+    if (lhs is rhs)
+        return 0;
+    // Regard null references as always being "less than"
+    if (!lhs)
+        return -1;
+    if (!rhs)
+        return 1;
+    return lhs.opCmp(rhs);
+}
+
+int __cmp(T)(const T[] lhs, const T[] rhs) @trusted
```

Comment:
@CyberShadow take that back, I decided to implement the whole shebang anyway, it doesn't take much code.